pkg/apis/core/validation/validation.go

```diff
@@ -421,6 +421,18 @@ func ValidateVolumes(volumes []core.Volume, podMeta *metav1.ObjectMeta, fldPath
 	return vols, allErrs
 }
 
+func ValidateVolumesUpdate(volumes, oldVolumes []core.Volume, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
```

Comment:
I think old and new should be correlated by Volume.Name and then an update check done on matching names to confirm the limits aren't set only in that case.
[jlsong01] Agree and fixed.

IIUC for other volumes, the limit check is verified in the non-update validation (?---can you confirm? Maybe that doesn't work correctly for existing volumes that have the limits set (which we want to allow)).
[jlsong01] the update validation for existing volumes that have the limits set should be hanled. 
validateVolumeSourceUpdate -> validateEphemeralVolumeSourceUpdate -> ValidatePersistentVolumeClaimTemplateUpdate -> ValidatePersistentVolumeClaimSpecUpdate
in ValidatePersistentVolumeClaimSpecUpdate(), the validation fails only in case existing volumes has no limits set while the new volumes has the limits. All the other cases are passed, including existing volumes that have the limits set.

Reply:
Yeah, check the number of new volumes against the old one is not correct. I have modified it to correlate Volume.Name of old and new volumes and do the udpate check accordingly.

You are right, for other volumes it doesn't work correctly for existing volumes that have the limits set. I have fixed ths problem.